server/src/main/java/org/opensearch/rest/BaseRestHandler.java

```diff
@@ -200,6 +202,32 @@ protected final String unrecognized(
         return Collections.emptySet();
     }
 
+    /**
+     * Parse the deprecated request parameter 'master_timeout', and add deprecated log if the parameter is used.
+     * It also validates whether the value of 'master_timeout' is the same with 'cluster_manager_timeout'.
+     * Remove the method along with MASTER_ROLE.
+     * @deprecated As of 2.0, because promoting inclusive language.
+     * @param mnr the action request
+     * @param request the REST request to handle
+     * @param logger the logger that logs deprecation notices
+     * @param logMsgKeyPrefix the key prefix of a deprecation message to avoid duplicate messages.
+     */
+    @Deprecated
+    protected static void parseDeprecatedMasterTimeoutParameter(
+        MasterNodeRequest mnr,
+        RestRequest request,
+        DeprecationLogger logger,
+        String logMsgKeyPrefix
+    ) {
+        final String MASTER_TIMEOUT_DEPRECATED_MESSAGE =
+            "Deprecated parameter [master_timeout] used. To promote inclusive language, please use [cluster_manager_timeout] instead. It will be unsupported in a future major version.";
+        if (request.hasParam("master_timeout")) {
+            logger.deprecate(logMsgKeyPrefix + "_master_timeout_parameter", MASTER_TIMEOUT_DEPRECATED_MESSAGE);
+            request.validateParamValuesAreEqual("master_timeout", "cluster_manager_timeout");
```

Comment:
Can we be more strict here and fail if both are set? Is there a use case where a user will need to specify both old and new?

Reply:
Sure, I can make the parameter handler more strict as you said, and then I can even remove the method `validateParamValuesAreEqual()`, so that make the code simpler.
There is no use case, but gives the user more freedom.

Reply:
Updated to allow only either parameter to be set. 👍